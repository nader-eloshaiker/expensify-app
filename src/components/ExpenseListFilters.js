import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setEndDate, setStartDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({
            calendarFocused
        }));
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    };

    onSortChange = (e) => {
        switch (e.target.value) {
            case 'date':
                this.props.sortByDate();
                break;
            case 'amount':
                this.props.sortByAmount();
                break;
            default:
                this.props.sortByDate();
        };
    };

    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={this.onTextChange}/>
                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => { dispatch(setTextFilter(text)) },
    sortByDate: () => { dispatch(sortByDate()) },
    sortByAmount: () => { dispatch(sortByAmount()) },
    setStartDate: (date) => { dispatch(setStartDate(date)) },
    setEndDate: (date) => { dispatch(setEndDate(date)) }
});

const mapStateToProps = (state) => ({
    filters: state.filters
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);