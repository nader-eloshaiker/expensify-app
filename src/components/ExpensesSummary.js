import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import numeral from 'numeral';


export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
    <div>
        <p>{expensesCount} {expensesCount === 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal/100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    const expensesCount = expenses ? expenses.length : 0;
    const expensesTotal = selectExpensesTotal(expenses);

    return {
        expensesCount,
        expensesTotal
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
