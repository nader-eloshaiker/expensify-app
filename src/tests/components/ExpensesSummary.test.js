import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render the expenses summary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} ExpensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render the expenses summary correctly with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={5} ExpensesTotal={235123} />);
    expect(wrapper).toMatchSnapshot();
});