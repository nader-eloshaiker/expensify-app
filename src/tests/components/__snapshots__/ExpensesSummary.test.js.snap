// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expenses summary correctly with 1 expense 1`] = `
<div>
  <p>
    1
     
    expense
     totalling 
    $0.00
  </p>
</div>
`;

exports[`should render the expenses summary correctly with many expenses 1`] = `
<div>
  <p>
    5
     
    expenses
     totalling 
    $0.00
  </p>
</div>
`;
